pkg/registry/apis/datasource/register.go

```diff
@@ -58,24 +58,28 @@ func RegisterAPIService(
 	accessControl accesscontrol.AccessControl,
 	reg prometheus.Registerer,
 ) (*DataSourceAPIBuilder, error) {
+	// We want to expose just a limited set of plugins
+	somePlugins := features.IsEnabledGlobally(featuremgmt.FlagDatasourceAPIServers)
+
 	// This requires devmode!
-	if !features.IsEnabledGlobally(featuremgmt.FlagGrafanaAPIServerWithExperimentalAPIs) {
+	if !(somePlugins || features.IsEnabledGlobally(featuremgmt.FlagGrafanaAPIServerWithExperimentalAPIs)) {
 		return nil, nil // skip registration unless opting into experimental apis
 	}
 
 	var err error
 	var builder *DataSourceAPIBuilder
 	all := pluginStore.Plugins(context.Background(), plugins.TypeDataSource)
+
 	// ATTENTION: Adding a datasource here requires the plugin to implement
-	// an AdmissionHandler to validate the datasource settings.
+	// an AdmissionHandler to validate the datasource settings. ??? Once we support save, yes but not now???
 	ids := []string{
 		"grafana-testdata-datasource",
 		"prometheus",
 		"graphite",
 	}
 
 	for _, ds := range all {
-		if !slices.Contains(ids, ds.ID) {
+		if somePlugins && !slices.Contains(ids, ds.ID) {
```

Comment:
The problem is that, because of the comment above (need to implement AdmissionHandler), all plugins that doesn't implement it will fail to save new datasources. We should probably make the AdmissionHandler optional first?

Reply:
I'm forgetting how that is wired up

Reply:
Hymm logic here seems wrong https://github.com/grafana/grafana/pull/90038/files#diff-f7dc375d45524623c26b3345a6964c8be4752f71bdfe472e8e291a3a7bd43378R278 

The frontend code will know if it can talk to v1 or v2 -- the backend changing its preferred version does not mean the frontend now knows how to deal with the new version

Reply:
ahh ok i see what it is doing now -- the smallest change is to make sure v0alpha1 is ignored from that list.  We want the AdmissionHandler to be required once we set an explicit apiVersion, but everything should be able to be enabled without that